ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -562,21 +563,11 @@ deleteValue
   => ShakeExtras
   -> k
   -> NormalizedFilePath
-  -> STM ()
-deleteValue ShakeExtras{dirtyKeys, state} key file = do
+  -> STM [Key]
```

Comment:
what is this?

Reply:
The rule result cache, storing the rule result and its diagnostic.

Reply:
I mean, the return value. It's the dirty keys, we just need to document it.

Reply:
done